tatanka/tanka/swaps.go

```diff
@@ -103,3 +101,49 @@ type MarketParameters struct {
 	BaseID  uint32 `json:"baseID"`
 	QuoteID uint32 `json:"quoteID"`
 }
+
+const (
+	OrderExpiration    = time.Hour * 12
+	OrderUpdateVersion = 0
+)
+
+type OrderUpdate struct {
+	*Order
+	Version    uint8     `json:"version"`
+	Expiration time.Time `json:"expiration"`
+	Settled    uint64    `json:"settled"`
```

Comment:
This field is curious. I'm not sure we need to track this.

Reply:
Ok. I thought if an order was on the books, there would be two ways to update. Expiration and the amount left. Would we not update if some is sold. Would we just cancel this order and make a new one?

Reply:
Ah. Good point. I saw you added a `Nonce` field. Perhaps that should just serve as an order ID instead, and an order update could simply be 

```Go
type OrderUpdate struct {
    From         tanka.PeerID
    Nonce       uint64
    Qty           uint64
    Stamp       time.time
}
```

In which case, it wouldn't really be a nonce any more, just a order identifier. And I used `Stamp` instead of `Expiration` because I think the time-to-staleness should be a network constant, not something defined as part of the order. So an order would update the `Stamp`, restarting the clock.